src/scripts/main.js

```diff
@@ -1,3 +1,343 @@
+    }
+
+    checkGameOver();
+  }
+};
+
+function slideRight() {
+  if (!gameOver) {
+    for (let r = 0; r < rows; r++) {
+      let row = board[r];
+
+      row.reverse();
+      row = slide(row);
+      row.reverse();
+
+      board[r] = row;
+
+      for (let c = 0; c < columns; c++) {
+        const field
+          = document.getElementById(r.toString() + '-' + c.toString());
+        const num = board[r][c];
+
+        update(field, num);
+      }
+    }
+
+    if (isMoved) {
+      appendNewCell();
+      isMoved = false;
+    }
+
+    checkGameOver();
+  }
+};
+
+function slideUp() {
+  if (!gameOver) {
+    for (let c = 0; c < columns; c++) {
+      let row = [board[0][c], board[1][c], board[2][c], board[3][c]];
+
+      row = slide(row);
+
+      for (let r = 0; r < rows; r++) {
+        board[r][c] = row[r];
+
+        const field
+          = document.getElementById(r.toString() + '-' + c.toString());
+        const num = board[r][c];
+
+        update(field, num);
+      }
+    }
+
+    if (isMoved) {
+      appendNewCell();
+      isMoved = false;
+    }
+
+    checkGameOver();
+  }
+};
+
+function slideDown() {
+  if (!gameOver) {
+    for (let c = 0; c < columns; c++) {
+      let row = [board[0][c], board[1][c], board[2][c], board[3][c]];
+
+      row.reverse();
+      row = slide(row);
+      row.reverse();
+
+      for (let r = 0; r < rows; r++) {
+        board[r][c] = row[r];
+
+        const field
+          = document.getElementById(r.toString() + '-' + c.toString());
+        const num = board[r][c];
+
+        update(field, num);
+      }
+    }
+
+    if (isMoved) {
+      appendNewCell();
+      isMoved = false;
+    }
+
+    checkGameOver();
+  }
+};
+
+function checkGameOver() {
+  let lose = false;
+  let win = false;
+  const zeros = [];
+
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < columns; c++) {
+      if (board[r][c] === 2048) {
+        win = true;
+      }
+
+      if (board[r][c] === 0) {
+        zeros.push(0);
+      }
+    }
+  }
+
+  for (let r = 0; r < rows - 1; r++) {
+    for (let c = 0; c < columns - 1; c++) {
+      const currentValue = board[r][c];
+
+      if (
+        (r > 0 && board[r - 1][c] === currentValue)
+        || (r < rows - 1 && board[r + 1][c] === currentValue)
+        || (c > 0 && board[r][c - 1] === currentValue)
+        || (c < columns - 1 && board[r][c + 1] === currentValue)
+      ) {
+        lose = false;
+        break;
+      } else {
+        lose = true;
+      }
+    }
+
+    if (!lose) {
+      break;
+    }
+  }
+
+  if (lose && zeros.length === 0) {
+    loseMessage.classList.remove('hidden');
+    gameOver = true;
+
+    return;
+  }
+
+  if (win) {
+    winMessage.classList.remove('hidden');
+    gameOver = true;
+  }
+}
+
+startButton.addEventListener('click', () => {
+  startGame();
+});
+
+restartButton.addEventListener('click', () => {
+  startGame();
+});
+
+document.addEventListener('keyup', (e) => {
+  if (e.code === 'ArrowLeft') {
```

Comment:
Move all directions to the object and use them like:

`e.code === Directions.ArrowLeft`